### PR TITLE
Add simple save model test

### PR DESCRIPTION
**Pull request by jeswan**
This PR closes https://github.com/nyu-mll/jiant/issues/1222 by adding save model tests for the simple runscript. These tests check the model saving behavior by:

1. Overfitting a model and checking the last model weights are different than the best model weights
2. Checking that only the best model is saved with the `do_save_best` argument.
3. Checking that only the last model is saved with the `do_save_last` argument.

This PR also corrects a small bug in saving behavior (required for the tests to pass).

### CODE REVIEW

File: tests/proj/simple/test_runscript.py [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
 
 
+def eq_state_dicts(model_1, model_2):
+    models_differ = 0
+    for key_item_1, key_item_2 in zip(model_1.items(), model_2.items()):
+        if torch.equal(key_item_1[1], key_item_2[1]):
+            pass
+        else:
+            return False
+    return True
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:17]: @zphang Do you think this belongs in a utils.py file?

Comment (zphang) [reply tests/proj/simple/test_runscript.py:17]: Yes, but (1) change the variable names to be generic (`state_dict_1`), and (2) document to emphasize that this is checking the values, not the reference/data pointers.

File: tests/proj/simple/test_runscript.py [hunk 2]
```diff
@@ -29,3 +31,83 @@ def test_simple_runscript(tmpdir, task_name, model_type):
...
+    run_name = f"{test_simple_runscript.__name__}_{task_name}_{model_type}_save_best"
+    args = run.RunConfiguration(
+        run_name=run_name,
+        exp_dir=exp_dir,
+        data_dir=data_dir,
+        model_type=model_type,
+        tasks=task_name,
+        train_examples_cap=16,
+        train_batch_size=16,
+        do_save_best=True,
```
Comment (zphang) [inline tests/proj/simple/test_runscript.py:85]: If this is just a saving test, we can limit the number of training steps.

**Review by zphang — APPROVED** [review]
Left a comment, but I'm fine merging for now.

File: tests/proj/simple/test_runscript.py [hunk 3]
```diff
@@ -29,3 +31,83 @@ def test_simple_runscript(tmpdir, task_name, model_type):
...
+    assert not torch_utils.eq_state_dicts(best_model_weights, last_model_weights)
+
+    run_name = f"{test_simple_runscript.__name__}_{task_name}_{model_type}_save_best"
+    args = run.RunConfiguration(
+        run_name=run_name,
+        exp_dir=exp_dir,
+        data_dir=data_dir,
+        model_type=model_type,
+        tasks=task_name,
+        train_examples_cap=16,
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:83]:
```suggestion
        max_steps=1,
```

File: tests/proj/simple/test_runscript.py [hunk 4]
```diff
@@ -29,3 +31,83 @@ def test_simple_runscript(tmpdir, task_name, model_type):
...
+
+    # check output last model
+    run_name = f"{test_simple_runscript.__name__}_{task_name}_{model_type}_save_last"
+    args = run.RunConfiguration(
+        run_name=run_name,
+        exp_dir=exp_dir,
+        data_dir=data_dir,
+        model_type=model_type,
+        tasks=task_name,
+        train_examples_cap=16,
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:103]:
```suggestion
        max_steps=1,
```

File: tests/proj/simple/test_runscript.py [hunk 5]
```diff
@@ -29,3 +31,83 @@ def test_simple_runscript(tmpdir, task_name, model_type):
...
+
+    downloader.download_data([task_name], data_dir)
+
+    args = run.RunConfiguration(
+        run_name=run_name,
+        exp_dir=exp_dir,
+        data_dir=data_dir,
+        model_type=model_type,
+        tasks=task_name,
+        train_examples_cap=128,
```
Comment (jeswan) [inline tests/proj/simple/test_runscript.py:52]:
```suggestion
        max_steps=1,
```

**Pull request merged**